fujiji-server/__tests__/listing/testGetAllListings.js

```diff
@@ -0,0 +1,24 @@
+const request = require('supertest');
+const app = require('../../src/app');
+
+describe('Test get all listings', () => {
+  it('successfully get all listings by existing city', async () => {
+    const res = await request(app)
+      .get('/listing')
+      .query({ city: 'Winnipeg' });
+    expect(res.statusCode).toEqual(200);
+  });
+
+  it('successfully get all listings by existing province', async () => {
+    const res = await request(app)
+      .get('/listing')
+      .query({ provinceCode: 'MB' });
+    expect(res.statusCode).toEqual(200);
+  });
+
+  it('undefined location returns expected error', async () => {
+    const res = await request(app)
+      .get('/listing');
+    expect(res.statusCode).toEqual(400);
+  });
+});
```

Comment:
How do we get the data for the test? Do we need to populate a separate database with test data?

Might be worth it looking into creating a setup file to create a test DB that contain test data. This definitely does not have to be included in this PR. Since this setup in itself might be pretty complex. Or we can always mock the response of the DB query lol (this is significantly easier than setting up separate db for testing)

Reply:
If we have a test db, we can populate that using beforeAll and then clean up that db using afterAll.